plugin/elasticity/cable.cc

```diff
@@ -50,28 +73,91 @@ void QuatDiff(mjtNum* quat, const mjtNum body_quat[4],
 //     scl         - scaling of the force
 //   outputs:
 //     qfrc        - local torque contribution
-void LocalForce(mjtNum qfrc[3], const mjtNum stiffness[4],
-                const mjtNum quat[4], const mjtNum omega0[3],
-                const mjtNum xquat[4], mjtNum scl) {
-  mjtNum omega[3], lfrc[3];
-
-  // compute curvature
-  mju_quat2Vel(omega, quat, scl);
-
-  // subtract omega0 in reference configuration
-  mjtNum tmp[] = {
-      - stiffness[0]*(omega[0] - omega0[0]) / stiffness[3],
-      - stiffness[1]*(omega[1] - omega0[1]) / stiffness[3],
-      - stiffness[2]*(omega[2] - omega0[2]) / stiffness[3],
-  };
-
+void LocalForce(mjtNum qfrc[3], mujoco::plugin::elasticity::Cable::stiffness_ Sel, mujoco::plugin::elasticity::Cable::stiffness_consts_ Sconst,
+    const mjtNum quat[4], mjtNum omega0[3], mjtNum last_stress[3], mjtNum last_strain[3], int userdatatype, mjtNum*  userdata, const mjtNum xquat[4], mjtNum scl) {
+    
+    mjtNum omega[3], lfrc[3];
+    
+    // compute curvature
+    mju_quat2Vel(omega, quat, scl);
+
+    // change of curvature from reference configuration
+    mjtNum d_omega[] = {
+        omega[0] - omega0[0],
+        omega[1] - omega0[1],
+        omega[2] - omega0[2]
+    };
+   
+    //actual strain
+    mjtNum strain[] = {
+         d_omega[0] / (Sel.L_Dyz + omega0[0]),
+         d_omega[1] / (Sel.L_Dy + omega0[1]),
+         d_omega[2] / (Sel.L_Dz + omega0[2]),
+    };
+    
+    mjtNum Yield_stress_top[] = {
```

Comment:
Could you document this plastic model with e.g. a paper or a link and/or a more detailed explanation in the code?